code/0.5_resmaple_rasters.R

```diff
@@ -0,0 +1,186 @@
+  ## -----------------------------------------------
+  resampled_raster = resample(weights_raster, clim_raster, method="bilinear")
+  
+  ## Save the raster
+  ## -----------------------------------------------
+  path = file.path(data_folder, "int", "rasterweights")
+  if(!dir.exists(path)){
+    
+    # If no - create it
+    message(crayon::yellow(paste('Creating', path)))
+    dir.create(path, recursive=T)
+    
+  }
+  
+  saveRDS(resampled_raster, file = file.path(path, paste0(data_source, "_", names(weights_raster), "_", extent, ".tif")))
+  
+  ## Make a data.table of the values of the resampled raster with lat/lon
+  ## -----------------------------------------------
+  weight_table <- as.data.frame(resampled_raster, xy=TRUE)
+  colnames(weight_table) <- c("x", "y", "weight")
+  
+  ## Save the weights table
+  ## -----------------------------------------------
+  fwrite(weight_table, file = file.path(path, paste0(data_source, "_", names(weights_raster), "_", extent, ".csv")))
+  
+  ## Return the weights table
+  ## -----------------------------------------------
+  return(weight_table)
+  
+}
+
+## Test function 
+## -----------------------------------------------
+
+read_potapov <- function(quad, year){
+  require(pacman)
+  pacman::p_load(here, raster)
+  # data_folder = here::here("data")
+  # data_folder = '/home/tcarleton/Climate/data'
+  crop_path <- file.path(data_folder, "raw", "weights", "cropland")
+  return(raster(file.path(crop_path, paste0("cropland_", quad, "_", year, ".tif"))))
+}
+
+# baby trial
+# p <- read_potapov(quad="NW", year=2011)
+# e <- extent(c(-1, 0, 9, 10))
+# calc_raster_weights(data_source = 'era5',  weights_raster = p, extent = e)
+
+# bigger test
+# us_counties <- tigris::counties() #Input polygons for testing
+# us_extent <- extent(us_counties)
+# calc_raster_weights(data_source = 'era5',  weights_raster = p, extent = us_extent)
+# p <- read_potapov(quad="NW", year=2011)
+# calc_raster_weights(data_source = 'era5',  weights_raster = p, extent = "full")
+
+
+# full run
+library(parallel)
+
+years <- c(2007, 2015, 2011, 2003, 2019)
+quads <- c("NW", "NE", "SW", "SE")
+
+full_potapov <- function(year){
+  library(data.table)
+  
+  rasters <- lapply(quads, read_potapov, year = year)
+  
+  no_cores <- detectCores() # Calculate the number of cores. Leave one in case something else needs to be done on the same computer at the same time. 
+  paste("working on year", year, "with", no_cores, "cores")
+  cl <- makeCluster(no_cores, type="FORK") # Initiate cluster. "FORK" means bring everything in your current environment with you. 
+  
+  tables <- parLapply(cl, rasters, calc_raster_weights, data_source = 'era5', extent = "full")
+  
+  stopCluster(cl)
+  
+  full_table <- rbindlist(tables)
+  # save the whole world of potapov data
+  path = file.path(data_folder, "int", "rasterweights")
+  fwrite(full_table, file = file.path(path, paste0("era5_cropland_", year, "_full.csv")))
```

Comment:
@anna-boser do we want to hardcode era5_cropland in here? isn't this resampling supposed to work with non-cropland weights (eg population)?